src/frontend/src/binder/expr/function.rs

```diff
@@ -166,7 +168,41 @@ impl Binder {
                 return Ok(ExprImpl::literal_varchar(self.db_name.clone()));
             }
             "current_schema" if inputs.is_empty() => {
-                return Ok(ExprImpl::literal_varchar(DEFAULT_SCHEMA_NAME.to_string()));
+                let first_schema_name = self
+                    .catalog
+                    .first_valid_schema(
+                        &self.db_name,
+                        &self.search_path,
+                        &self.auth_context.user_name,
+                    )
+                    .map(|schema| schema.name())
+                    .unwrap_or_default();
```

Comment:
When there are no valid schemas, PostgreSQL returns `null::name` rather than `''::name` (and we represent `name` with `varchar` in RW now):
```
test=# set search_path to non_exist;
SET
test=# select current_schema is null;
 ?column? 
----------
 t
(1 row)
```